Assets/MRTK/SDK/Experimental/Drawstring/Scripts/ElasticHandleLogic.cs

```diff
@@ -0,0 +1,123 @@
+        /// <param name="rightHandleStart">World position of "right" handle point.</param>
+        /// <param name="extentInfo">Properties of the linear 1-D elastic extent to be manipulated</param>
+        /// <param name="elasticProperties">Properties of the elastic material/spring.</param>
+        /// <param name="leftHandleVelocity">Optional, initial velocity in 1-dimensional stretch space</param>
+        /// <param name="rightHandleVelocity">Optional, initial velocity in 1-dimensional stretch space</param>
+        public virtual void Setup(Vector3 leftHandleStart, Vector3 rightHandleStart,
+                                    ElasticExtentProperties<float> extentInfo, ElasticProperties elasticProperties,
+                                    float leftHandleVelocity = 0.0f, float rightHandleVelocity = 0.0f)
+        {
+            isSetup = true;
+            leftInitialPosition = leftHandleStart;
+            rightInitialPosition = rightHandleStart;
+            elasticSystem = new LinearElasticSystem((leftHandleStart - rightHandleStart).magnitude, 0.0f, extentInfo, elasticProperties);
+        }
+
+        /// <summary>
+        /// Update the internal state of the damped harmonic oscillator, given the left and right pointer positions.
+        /// Note, both left and right input positions are nullable; set the pointer position to null
+        /// if only one of the handles is being interacted with. Returns the calculated distance between the
+        /// handles.
+        /// 
+        /// </summary>
+        /// <param name="leftPointer">World position of the pointer manipulating the left handle.</param>
+        /// <param name="rightPointer">World position of the pointer manipulating the right handle.</param>
+        /// <param name="deltaTime">Amount of time to simulate this step.</param>
+        /// <param name="normalizedAxis">World axis of the two-handle system.</param>
+        public virtual float Update(Vector3? leftPointer, Vector3? rightPointer, float deltaTime, Vector3? normalizedAxis = null)
+        {
+            // If we have not been Setup() yet
+            if (!isSetup) { return 0.1f; }
+
+            var handleAxis = normalizedAxis ?? (rightInitialPosition - leftInitialPosition).normalized;
+
+            var handDistance = elasticSystem.GetCurrentValue();
+            if (leftPointer.HasValue && rightPointer.HasValue)
+            {
+                // If we have both pointers, we simply project the vector between the
+                // two hands onto the handle axis vector.
+                handDistance = Vector3.Dot(rightPointer.Value - leftPointer.Value, handleAxis);
+            }
+            else if (!leftPointer.HasValue && rightPointer.HasValue)
+            {
+                // If we only have a right pointer, calculate the hand distance
+                // as twice the distance of the right pointer from the center.
```

Comment:
What's the significance of the 2x value? I assume this is to ensure it's more usable in one-handed scenarios. Anytime you use a hardcoded value it's important to specfy its significance.